kong/plugins/acme/storage/config_adapters/redis.lua

```diff
@@ -3,13 +3,13 @@ local function redis_config_adapter(conf)
         host = conf.host,
         port = conf.port,
         database = conf.database,
-        auth = conf.password or conf.auth, -- allow conf.auth until 4.0 version
+        auth = conf.password,
         ssl = conf.ssl,
         ssl_verify = conf.ssl_verify,
-        ssl_server_name = conf.server_name or conf.ssl_server_name, -- allow conf.ssl_server_name until 4.0 version
+        ssl_server_name = conf.server_name,
 
-        namespace = conf.extra_options.namespace or conf.namespace, -- allow conf.namespace until 4.0 version
-        scan_count = conf.extra_options.scan_count or conf.scan_count, -- allow conf.scan_count until 4.0 version
```

Comment:
Unfortunately this approach did not work because `conf.extra_options.scan_count` was populated with a default value o 10 so even if the user tried to configure the plugin using the old way it would get overwritten by the db default.